tests/codex/sales/testsales.nim

```diff
@@ -450,10 +450,35 @@ asyncchecksuite "Sales":
       return success()
     createAvailability()
     await market.requestStorage(request)
-    clock.set(request.expiry.truncate(int64))
+
+    # If we would not await, then the `clock.set` would run "too fast" as the `subscribeCancellation()`
+    # would otherwise not set the timeout early enough as it uses `clock.now` in the deadline calculation.
+    await sleepAsync(chronos.milliseconds(100))
```

Comment:
I understand the need to wait a bit here, but can't we wait on something specific? Maybe we can wait here until onStore is called, for instance?

Waiting for a number of milliseconds has two unfortunate side-effects:
1. You're probably waiting too long most of the time, slowing down the test run
2. You're probably waiting not long enough some of the time, making the test fail every now and then in CI

:smile:

Reply:
I agree with you, but I have not found a way to wait for something concrete. I have tried using the `onStore` callback, but even that is "too fast" and does not give enough time to properly set up the expiry timeouts. So the only possibility I see is adding a special callback called "onCancellationSet" or something similar, which could be used here. However, in my honest opinion, that is not worthwhile. Honestly, I have spent way too much time on this PR (again) and would like to move on to other tasks, so I would leave it as it is for now.

Reply:
Fair enough!